src/main/kotlin/za/ac/sun/plume/drivers/OverflowDbDriver.kt

```diff
@@ -0,0 +1,255 @@
+        return EvaluationStrategy.valueOf(str)
+    }
+
+    private fun convertDispatchType(str : String) : DispatchType {
+        return DispatchType.valueOf(str)
+    }
+
+    private fun <T>getOrElse(opt : scala.Option<T>, def : T) : T {
+        class F<T>(val x : T) : AbstractFunction0<T>() {
+            override fun apply(): T {
+                return x;
+            }
+        }
+        return opt.getOrElse(F(def))
+    }
+
+    override fun exists(v: PlumeVertex): Boolean {
+        return (graph.node(v.hashCode().toLong()) != null)
+    }
+
+    override fun exists(fromV: PlumeVertex, toV: PlumeVertex, edge: EdgeLabel): Boolean {
+        val srcNode = graph.node(fromV.hashCode().toLong())
+        val dstNode = graph.node(toV.hashCode().toLong())
+        return srcNode != null && srcNode.out(edge.name).asSequence().toList().filter { node -> node.id().equals(dstNode.id()) }.isNotEmpty()
+    }
+
+    override fun addEdge(fromV: PlumeVertex, toV: PlumeVertex, edge: EdgeLabel) {
+        var srcNode = graph.node(fromV.hashCode().toLong())
+        if (srcNode == null) {
+            addVertex(fromV)
+            srcNode = graph.node(fromV.hashCode().toLong())
+        }
+        var dstNode = graph.node(toV.hashCode().toLong())
+        if (dstNode == null) {
+            addVertex(toV)
+            dstNode = graph.node(toV.hashCode().toLong())
+        }
+
+        try {
+            srcNode.addEdge(edge.name, dstNode)
+        } catch(exc : RuntimeException) {
+            throw PlumeSchemaViolationException(fromV, toV, edge)
+        }
+
+    }
+
+    override fun maxOrder(): Int {
+        return Traversals.maxOrder(graph)
+    }
+
+    override fun clearGraph(): IDriver {
+        Traversals.clearGraph(graph)
+        return this
+    }
+
+    override fun getWholeGraph(): PlumeGraph {
+        return nodesWithEdgesToPlumeGraph(Traversals.getWholeGraph(graph))
+    }
+
+    override fun getMethod(fullName: String, signature: String): PlumeGraph {
+        return edgeListToPlumeGraph(Traversals.getMethod(graph, fullName, signature))
+    }
+
+    override fun getMethod(fullName: String, signature: String, includeBody: Boolean): PlumeGraph {
+        if (includeBody) return getMethod(fullName, signature)
+        return edgeListToPlumeGraph(Traversals.getMethodStub(graph, fullName, signature))
+    }
+
+    private fun nodesWithEdgesToPlumeGraph(nodesWithEdges : List<Tuple2<StoredNode, List<Edge>>>) : PlumeGraph {
+        val plumeGraph = PlumeGraph()
+        val plumeVertices = nodesWithEdges
+                .map{ x -> x._1 }
+                .distinct()
+                .map{node -> Pair(node.id(), convert(node)) }
+                .toMap()
+        plumeVertices.values.forEach { v -> plumeGraph.addVertex(v) }
+        val edges = nodesWithEdges.flatMap { x -> x._2 }
+        edges.forEach { edge ->
+            val srcNode = plumeVertices.get(edge.outNode().id())
```

Comment:
Instead of using `get` you can use indexing operator e.g. `plumeVertices[edge.outNode().id()]`